Formula/apache-arrow-static.rb

```diff
@@ -42,6 +37,7 @@ def install
       -DARROW_PARQUET=ON
       -DARROW_BUILD_SHARED=OFF
       -DARROW_JEMALLOC=ON
+      -DARROW_MIMALLOC=ON
```

Comment:
FWIW I found that mimalloc didn't compile on my M1 mac, not sure if you've seen similar. So we may need to make this conditional.

Reply:
Havent' tried the R package, but building arrow seems to work.